proxy-attestation-server/src/attestation.rs

```diff
@@ -55,3 +86,148 @@ pub async fn start(body_string: String) -> ProxyAttestationServerResponder {
         _ => Err(ProxyAttestationServerError::UnknownAttestationTokenError),
     }
 }
+
+fn convert_csr_to_certificate(csr_der: &[u8]) -> Result<openssl::x509::X509, ProxyAttestationServerError> {
+    let csr = openssl::x509::X509Req::from_der(&csr_der)?;
+    // first, verify the signature on the CSR
+    let public_key = csr.public_key()?;
+    let verify_result = csr.verify(&public_key)
+        .map_err(|err| {
+            println!("proxy-attestation-server::attestation::convert_csr_to_certificate csr.verify failed:{:?}", err);
+            err
+        })?;
+
+    if !verify_result {
+        println!("proxy_attestation_server::convert_csr_to_certificate verify of CSR failed");
+        return Err(ProxyAttestationServerError::CsrVerifyError);
+    }
+    let mut cert_builder = openssl::x509::X509Builder::new()
+        .map_err(|err| {
+            println!("proxy-attestation-server::attestation::convert_csr_to_certificate X509Builder::new failed:{:?}", err);
+            err
+        })?;
+    cert_builder.set_version(2)
+        .map_err(|err| {
+            println!("proxy-attestation-server::attestation::convert_csr_to_certificate set_version failed:{:?}", err);
+            err
+        })?;
+    let now = openssl::asn1::Asn1Time::days_from_now(0)
+                .map_err(|err| {
+                    println!("proxy-attestation-server::attestation::convert_csr_to_certificate days_from_now failed:{:?}", err);
+                    err
+                })?;
+    cert_builder.set_not_before(&now)
+        .map_err(|err| {
+            println!("proxy-attestation-server::attestation::convert_csr_to_certificate set_not_before failed:{:?}", err);
+            err
+        })?;
+
+    // TODO: Make this configurable
+    let expiry =openssl::asn1::Asn1Time::days_from_now(1)
```

Comment:
Are you sure you want that? I just ran it, and there are a lot of formatting changes that are not related to this change (meaning that `cargo fmt` is not being run frequently) which would muddy up the specifics of this change quite a bit.

Reply:
Hmm, OK.  Maybe we need to just run a `cargo fmt` throughout and open up a separate PR...

Reply:
I think that would be best. 

Also, to prevent this problem in the future, we might want to add a check to CI that will reject any PR that `cargo fmt` causes changes on.